### PR TITLE
Use default apps from cluster chart

### PR DESCRIPTION
**Pull request by njuettner**
Towards: https://github.com/giantswarm/roadmap/issues/3462

Breaking change, it will move all apps to cluster-azure, so we can drop `default-apps-azure`

 **How to migrate to cluster-azure v0.14.0**

 * Update `<cluster name>` to the latest version of `default-apps-azure` (v0.14.0). 
 * In ConfigMap `<cluster name>-default-apps-userconfig` set `.Values.deleteOptions.moveAppsHelmOwnershipToClusterAzure` to `true`. 
 * Delete `<cluster name>-default-apps` App. 
 * Update `<cluster name>` to the latest version of `cluster-azure` (v0.14.0).

I added it in the README.

### Please check if PR meets these requirements

- [ ] Results of the diffs have been examined and no unintended changes are being introduced.

### Helper

* to disable the GH Action generating manifests diffs for installations, between target and source branches, comment the `/no_diffs_printing` on the PR.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

**Comment by njuettner** [comment]
I'm currently waiting for https://github.com/giantswarm/cluster-test-suites/pull/333 so I can run the E2e test but I already tested it manually.

This will allow us to opt out from `default-apps-azure` with release `v0.14.0`

File: helm/cluster-azure/templates/_externaldns_private_app.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{{/* Azure-specific external-dns app for private clusters */}}
```
Comment (njuettner) [inline helm/cluster-azure/templates/_externaldns_private_app.yaml:1]: External DNS Private is special for Azure Private clusters we install it separate.

**Comment by njuettner** [comment]
Unfortunately I can't really run the test via E2e 

I tried to tag my PR with `v0.14.0-alpha.1`
But the tests always add a hash

```
  version: v0.14.0-alpha.1-9b54bd565a8be77f52817ed9e16f0bc2ed40ca08
status:
  appVersion: ""
  release:
    lastDeployed: null
    reason: pulling chart `oci://giantswarmpublic.azurecr.io/cluster-catalog/cluster-azure:0.14.0-alpha.1-9b54bd565a8be77f52817ed9e16f0bc2ed40ca08`
      failed
    status: chart-pull-failed
  version: ""
```

Changing it manually for CAPZ standard it worked fine 
https://tekton.ci.giantswarm.io/#/namespaces/tekton-pipelines/pipelineruns/pr-cluster-azure-290-cluster-test-suitesr6fck?pipelineTask=run-tests&taskRunName=pr-cluster-azure-290-cluster-test-suitesr6fck-run-tests-0

However I need to have >=v0.14.0 (currently 0.13.0) so running it without tagging wouldn't work.

I tested it manually and it worked

**Comment by njuettner** [comment]
Important is we need this for v0.14.0, because I had to add logic in `clustertest` which verifies we unified our apps.

**Review by weseven — APPROVED** [review]

File: helm/cluster-azure/templates/_certmanager_app_config.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{{/* Azure-specific cilium Helm values*/}}
```
Comment (fiunchinho) [inline helm/cluster-azure/templates/_certmanager_app_config.yaml:1]:
```suggestion
{{/* Azure-specific cert-manager Helm values*/}}
```

File: helm/cluster-azure/templates/_cilium_helmrelease_config.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{{/* Azure-specific cilium Helm values*/}}
+{{/* https://github.com/giantswarm/cilium-app/blob/main/helm/cilium/values.yaml*/}}
+{{- define "azureCiliumHelmValues" }}
```
Comment (fiunchinho) [inline helm/cluster-azure/templates/_cilium_helmrelease_config.yaml:3]: how is this used later on to install `cilium`?

Comment (njuettner) [reply helm/cluster-azure/templates/_cilium_helmrelease_config.yaml:3]: it is installed via cluster chart, cluster chart contains all the apps which are installed for all provideers like cilium. the only thing which is different is the configuration of all apps. that's why we add the config and in value schema we enable apps and plus adding the name of to the configuration.

cluster chart will read the configuration, e.g. azureCiliumHelmValues and render the values in the app itself.

Comment (fiunchinho) [reply helm/cluster-azure/templates/_cilium_helmrelease_config.yaml:3]: ok, thank you!

**Review by fiunchinho — APPROVED** [review]
Looks good!

**Pull request merged**